src/lib/i18n.test.js

```diff
@@ -0,0 +1,52 @@
+/* global describe, it, expect, beforeAll */
+// @ts-check
+import i18n, { localesList } from '../i18n.js'
+import { getLanguage, getCurrentLanguage } from './i18n.js'
+import languages from './languages.json'
+import { readdir } from 'node:fs/promises'
+
+const testEachLanguage = (fn) => {
+  Object.keys(languages).forEach((lang) => fn(lang))
+}
+
+const allLanguages = (await readdir('./public/locales', { withFileTypes: true }))
+  .filter(dirent => dirent.isDirectory())
+  .map(dirent => dirent.name)
+
+describe('i18n', function () {
+  it('should have a languages.json entry for each folder', function () {
+    const namedLocales = localesList.map(({ locale }) => locale)
+    expect(namedLocales).toEqual(allLanguages)
+  })
```

Comment:
this will ensure our language selector is always up to date. i.e. this test fails if we haven't followed the steps from `/docs/LOCALIZATION.md` in the `Pulling translations from Transifex` section. specifically, the `npx -q @olizilla/lol public/locales > src/lib/languages.json` command.